src/main/java/org/stellar/sdk/MemoText.java

```diff
@@ -4,26 +4,33 @@
 import org.stellar.sdk.xdr.MemoType;
 
 import java.nio.charset.Charset;
+import java.util.Arrays;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Represents MEMO_TEXT.
  */
 public class MemoText extends Memo {
-  private String text;
+  private byte[] text;
 
   public MemoText(String text) {
-    this.text = checkNotNull(text, "text cannot be null");
+    this(checkNotNull(text, "text cannot be null").getBytes((Charset.forName("UTF-8"))));
+  }
 
-    int length = text.getBytes((Charset.forName("UTF-8"))).length;
-    if (length > 28) {
-      throw new MemoTooLongException("text must be <= 28 bytes. length=" + String.valueOf(length));
+  public MemoText(byte[] text) {
+    this.text = checkNotNull(text, "text cannot be null");
+    if (this.text.length > 28) {
+      throw new MemoTooLongException("text must be <= 28 bytes. length=" + String.valueOf(this.text.length));
     }
   }
 
   public String getText() {
-    return text;
+    return new String(text, Charset.forName("UTF-8"));
```

Comment:
Consider for consistency
```suggestion
    return new String(this.text, Charset.forName("UTF-8"));
```